File: software/backend/src/main/java/ru/scms/back/dto/SleeveDto.java
package ru.scms.back.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import ru.scms.back.enums.SleeveGender;
import ru.scms.back.enums.SleeveStatus;

public record SleeveDto(
        Long id,
        String code,
        SleeveGender gender,
        Integer height,
        Integer weight,
        Integer age,
        Long geneticArchiveId,
        String geneticArchiveName,
        SleeveStatus status,
        String dnaDonor,
        String notes,
        LocalDateTime cultivationStartedAt,
        LocalDate plannedReadyAt,
        Integer cultivationStagePercent,
        Long reservedForUserId,
        String reservedForUserName,
        LocalDateTime createdAt) {}

File: software/client-portal/src/shared/api/dto/sleeveDto.ts
import type { SleeveGender, SleeveStatus } from "./enums";

export interface SleeveDto {
  id: number;
  code: string;
  gender: SleeveGender;
  height: number;
  weight: number;
  age: number;
  geneticArchiveId: number | null;
  geneticArchiveName: string | null;
  status: SleeveStatus;
  dnaDonor: string | null;
  notes: string | null;
  cultivationStartedAt: string | null;
  plannedReadyAt: string | null;
  cultivationStagePercent: number;
  reservedForUserId: number | null;
  reservedForUserName: string | null;
  createdAt: string;
}
